src/System.Private.CoreLib/shared/System/Globalization/CompareInfo.cs

```diff
@@ -1041,6 +1041,16 @@ internal int IndexOfOrdinal(ReadOnlySpan<char> source, ReadOnlySpan<char> value,
             Debug.Assert(!GlobalizationMode.Invariant);
```

Comment:
Are we ever come here with `ignoreCase: true` (with the other upfront special casing)? Maybe this can be renamed to IndexOfOrdinalIgnoreCase.

Reply:
I'm not entirely sure what the difference is between the two calls:
```csharp
CompareInfo.InvariantIndexOf(span, value, ignoreCase: true)
CompareInfo.Invariant.IndexOfOrdinal(span, value, ignoreCase: true)
```
Other than the first is used when `GlobalizationMode.Invariant`

Reply:
Changed anyway

Reply:
i.e. is `GlobalizationMode.Invariant && GetCaseCompareOfComparisonCulture(comparisonType) != CompareOptions.None` the same as `Ordinal` so could it take the faster path also?

Reply:
Ah InvariantCulture 😄